pkg/registry/pod/strategy.go

```diff
@@ -65,6 +65,21 @@ func (podStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newPod := obj.(*api.Pod)
 	oldPod := old.(*api.Pod)
 	newPod.Status = oldPod.Status
+	// TODO: VolumeMounts of init containers currently aren't merged properly
+	// by the PATCH handler because they are stored as a string under
+	// PodInitContainersAnnotationKey (pod.alpha.kubernetes.io/init-containers)
+	// in annotations of v1.PodSpec.
+	// In the existing Pod, VolumeMounts of init containers may be altered by
+	// secrets / config maps mechanisms causing ValidatePodUpdate to fail.
+	// As a workaround, changes in VolumeMounts of init containers are
+	// currently ignored and don't cause validation errors.
+	// Remove this workaround when InitContainers go out of alpha and
+	// PodInitContainersAnnotationKey is removed.
+	if len(oldPod.Spec.InitContainers) == len(newPod.Spec.InitContainers) {
```

Comment:
I can remove this workaround from PR but this way it will be a semi-fix for now.